crypto-pgp/src/androidTest/kotlin/dev/msfjarvis/aps/crypto/GPGKeyManagerTest.kt

```
@@ -0,0 +1,151 @@
+import com.github.michaelbull.result.getError
+import com.github.michaelbull.result.unwrap
+import com.proton.Gopenpgp.crypto.Key
+import dev.msfjarvis.aps.crypto.utils.CryptoConstants
+import dev.msfjarvis.aps.cryptopgp.test.R
+import dev.msfjarvis.aps.data.crypto.GPGKeyManager
+import dev.msfjarvis.aps.data.crypto.GPGKeyPair
+import java.io.File
+import java.lang.IllegalStateException
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.After
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+public class GPGKeyManagerTest {
+
+  private val testCoroutineDispatcher = TestCoroutineDispatcher()
+
+  @After
+  public fun tearDown() {
+    val filesDir = getFilesDir()
+    val keysDir = File(filesDir, GPGKeyManager.KEY_DIR_NAME)
+
+    keysDir.deleteRecursively()
+  }
+
+  @Test
+  public fun testAddingKey() {
+    runBlockingTest {
+      val gpgKeyManager = GPGKeyManager(getFilesDir().absolutePath, testCoroutineDispatcher)
+      val key = GPGKeyPair(Key(getKey()))
+
+      // Check if the key id returned is correct
+      val keyId = gpgKeyManager.addKey(key).unwrap()
+      assertEquals(CryptoConstants.KEY_ID, keyId)
+
+      // Check if the keys directory have one file
+      val keysDir = File(getFilesDir(), GPGKeyManager.KEY_DIR_NAME)
+      assertEquals(1, keysDir.list()?.size)
+
+      // Check if the file name is correct
+      val keyFile = keysDir.listFiles()?.first()
+      assertEquals(keyFile?.name, "$keyId.${GPGKeyManager.KEY_EXTENSION}")
+    }
+  }
+
+  @Test
+  public fun testRemovingKey() {
+    runBlockingTest {
+      val gpgKeyManager = GPGKeyManager(getFilesDir().absolutePath, testCoroutineDispatcher)
+      val key = GPGKeyPair(Key(getKey()))
+
+      // Add key using KeyManager
+      gpgKeyManager.addKey(key)
```

Call `unwrap` here to ensure errors don't get swallowed.